test/helpers.py

```diff
@@ -89,7 +89,9 @@ def _packages_from_json(env_export):
         """Extract packages and versions from the lines returned by the list of specifications"""
         dependencies = json.loads(env_export).get("dependencies")
         packages_dict = dict()
-        for split in map(lambda x: x.split("=", 1), dependencies):
+        for split in map(lambda x: x.split("=", 1) if isinstance(x, str) else None, dependencies):
+            if split is None:
+                break
```

Comment:
This code is very hard to read for me! I'd like to make it more readable.

Do you want to abort the looping (`break`) on first encountering a non string item in `dependencies`, or that you want to ignore it (`continue`)?

If you want to ignore some non-string dependencies, then i suggest we pre-process `dependencies` by filtering out those instead.

I'm a bit vary of seeing this change without understanding what situation this happens in. What value is it that we can end up seeing in the `dependencies` list?

Reply:
You're right this fix is awful and buggy, so thank you for the review 🐞.
I've replaced it by some more readable code.

```python
# Filtering packages installed through pip in this case it's a dict {'pip': ['toree==0.3.0']}
# Since we only manage packages installed through conda here
dependencies = filter(lambda x:  isinstance(x, str), dependencies)
for split in map(lambda x: x.split("=", 1), dependencies):
[...]
```

So I expect to have in `dependencies` only `conda` packages and so I can skip packages installed through `pip` that are reported as a `dict` (`{'pip': ['toree==0.3.0']}`).